binding/react-native/android/src/main/java/ai/picovoice/reactnative/porcupine/PorcupineModule.java

```diff
@@ -44,28 +44,48 @@ public String getName() {
     return "PvPorcupine";
   }
 
-   @Override
-  public Map<String, Object> getConstants() {
-    // default model file
-    final File resourceDirectory = reactContext.getFilesDir();
-    final Map<String, Object> constants = new HashMap<>();
-    constants.put("DEFAULT_MODEL_PATH", new File(resourceDirectory, "porcupine_params.pv").getAbsolutePath());
-
-    // default keyword files
-    final Map<String, String> keywordPaths = new HashMap<>();
-    for (Porcupine.BuiltInKeyword x : Porcupine.BuiltInKeyword.values()) {
-      String fileName = x.name().toLowerCase();
-      String keyword = fileName.replace('_', ' ');
-      keywordPaths.put(keyword, new File(resourceDirectory, fileName + ".ppn").getAbsolutePath());
+  @ReactMethod
+  public void fromBuiltInKeywords(String accessKey, String modelPath, ReadableArray keywords, ReadableArray sensitivities, Promise promise) {
+    // convert from ReadableArrays to Java types
+    Porcupine.BuiltInKeyword[] keywordsJava = new Porcupine.BuiltInKeyword[keywords.size()];
+    for (int i = 0; i < keywords.size(); i++) {
+      try {
+        keywordsJava[i] = Porcupine.BuiltInKeyword.valueOf(keywords.getString(i).toUpperCase());
+      } catch (IllegalArgumentException e) {
+        promise.reject(
+                PorcupineInvalidArgumentException.class.getSimpleName(),
+                String.format("'%s' is not a built-in keyword", keywords.getString(i)));
+        return;
+      }
     }
-    constants.put("KEYWORD_PATHS", keywordPaths);
 
-    return constants;
+    float[] sensitivitiesJava = new float[sensitivities.size()];
+    for (int i = 0; i < sensitivities.size(); i++) {
+      sensitivitiesJava[i] = (float) sensitivities.getDouble(i);
+    }
+
+    try {
+      Porcupine porcupine = new Porcupine.Builder()
+                              .setAccessKey(accessKey)
+                              .setModelPath(modelPath.length() == 0 ? null : modelPath)
+                              .setKeywords(keywordsJava)
```

Comment:
is it possible that they pass in an empty array of keywords? if so we should probably also do a `keywordsJava.length == 0 ? null : keywordsJava`